Mathlib/Algebra/Order/Monoid/CharZero.lean

```diff
@@ -0,0 +1,32 @@
+/-
+Copyright (c) 2016 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Leonardo de Moura, Mario Carneiro
+-/
+import Mathlib.Algebra.CharZero.Defs
+import Mathlib.Algebra.Order.Monoid.Defs
+import Mathlib.Algebra.Order.Monoid.NatCast
+
+#align_import algebra.order.ring.char_zero from "leanprover-community/mathlib"@"655994e298904d7e5bbd1e18c95defd7b543eb94"
+
+/-!
+# Strict ordered additive commutative monoids with one have characteristic zero
+
+An ordered additive commutative monoid with one in which addition is strictly monotone and `0 ≠ 1`
+has characteristic zero. This includes `StrictOrderedSemiring`s as well as `StarOrderedRing R` where
+`R` is a `Ring`.
```

Comment:
Can you make clear here that `StarOrderedRing R` doesn't imply `StrictOrderedSemiring`? It's currently not clear to me why.

Reply:
Done, but see also #7309